Sources/NIO/System.swift

```diff
@@ -310,10 +310,22 @@ internal enum Posix {
             let fd = sysAccept(descriptor, addr, len)
 
             #if !os(Linux)
-                if fd != -1 {
-                    let ret = try? Posix.fcntl(descriptor: fd, command: F_SETNOSIGPIPE, value: 1)
-                    assert(ret == .some(0),
```

Comment:
Is there any reason this assertion went away? Given that the error was throwing, I'd assume that the assertion could stay there.

Reply:
The assertion is redundant once `try?` was replaced with `try` - `fcntl()` with `F_SETFL` will only ever return `0` (success) or `-1` (error). In the old logic, errors were being eaten anyway (with the side effect that you could only see _what_ error occurred by inspecting _errno_ in the debugger) so it was just "assert that no error occurred". With the new logic, any error results in an `IOError` being thrown, so the only remaining return value possible is `0`, and the assertion is moved into the `catch` block so we can treat `EINTR` as non-crashing.

I suppose there's an argument that you could still assert that the non-error return value really was zero, but at that point you're checking for either the postcondition of `wrapSyscall {}` (which should be unit-tested directly, not asserted by callers) or the documented behavior of `fcntl(F_SETFL)`, which is an endless rabbit hole anyway because if the syscalls start returning nonsense results you probably have a larger issue than a failed assertion 🙂 (Not to  mention this particular code only runs on macOS anyway and we really ought to know the behavior there!)